controller/server/view/network_details_page.ml

```diff
@@ -45,14 +45,14 @@ let not_connected_form service =
               ~a:[ a_label_for passphrase_id ]
               [ txt "Passphrase" ]
           ]
-      ; input
-          ~a:[ a_input_type `Text
-          ; a_class [ "d-Input";  "d-Network__Input" ]
+      ; Unsafe.node "password-input"
```

Comment:
I see unsafe, I scream :scream: 
Just for sake of clarity, why wasn't that implementable with just `input`?

Reply:
That’s what I initially did in https://github.com/guyonvarch/playos/commit/53c87c741ca07087a9bf38628574c896a7edd1e9#diff-15ac6edb2f3a04dc0bf427b2e2d64b6716f8c9af099237aeb36b347e7a300d29R49.

But using an input, you can’t add other elements in the DOM under it. So, the workaround was to create a `span` block, replacin the `input`, and adding the `input` as a child: https://github.com/guyonvarch/playos/commit/53c87c741ca07087a9bf38628574c896a7edd1e9#diff-5a9bbb8656c0233df5590430531857fbf4dd67ed38d5e4233388b8c3aa8e6faaR44-R51.

But the current solution don’t work with progressive enhancement. In case JS is disabled, it’s just a div, potentially with some style:

![2021-05-11_09-18-51](https://user-images.githubusercontent.com/6768842/117774580-f8ae8900-b239-11eb-964d-72cbb36e3ca4.png)

So, I am wondering if the hack was finally not that bad. Or another solution is to provide a wrapper component around an input.

Reply:
Going back to the progressive enhancement method: https://github.com/dividat/playos/pull/69/commits/9df73828cb7d6ddcc09392961270d03e48c4ddf8#.

There is the “hack” I mentioned, but the previous method also had some non trivial code to make it work, moving the attributes to the input:
https://github.com/dividat/playos/pull/69/commits/9df73828cb7d6ddcc09392961270d03e48c4ddf8#diff-5a9bbb8656c0233df5590430531857fbf4dd67ed38d5e4233388b8c3aa8e6faaL44-L48

The wrapper solution, which I did not tried, may be the saner implementation, but this would be more cumbersome to use.